src/attendance/components/AttendanceSetup.tsx

```diff
@@ -28,48 +25,44 @@ export const AttendanceSetup: React.FC = () => {
     setAnchorEl(null);
   };
 
-  const handleUpdated = () => { removeEditors(); loadData(); }
+  const handleUpdated = () => { removeEditors(); refetch(); }
   const selectCampus = (campus: CampusInterface) => { removeEditors(); if (campus.name !== "Undefined") setSelectedCampus(campus); }
   const selectService = (service: ServiceInterface) => { removeEditors(); setSelectedService(service); }
   const selectServiceTime = (service: ServiceTimeInterface) => { removeEditors(); setSelectedServiceTime(service); }
   const removeEditors = () => { setSelectedCampus(null); setSelectedService(null); setSelectedServiceTime(null); }
-
-  const loadData = () => {
-    ApiHelper.get("/attendancerecords/tree", "AttendanceApi").then(data => {
-      if(isMounted()) {
-        setAttendance(data);
-      }
-    })
-    ApiHelper.get("/groupservicetimes", "AttendanceApi").then(data => {
-      if(isMounted()) {
-        setGroupServiceTimes(data);
-      }
-    })
-    ApiHelper.get("/groups", "MembershipApi").then(data => {
-      if(isMounted()) {
-        setGroups(data);
-      }
-    })
-  }
+  const refetch = () => { attendance.refetch(); groupServiceTimes.refetch(); groups.refetch(); }
+  
+  const attendance = useQuery<AttendanceInterface[]>({
+    queryKey: ["/attendancerecords/tree", "AttendanceApi"],
+    placeholderData: [],
+  });
```

Comment:
useQuery, what we will hopefully use for all data fetching.
These are using the default queryFn defined below but I could pass a data loading function here like.
```typescript
  const attendance = useQuery<AttendanceInterface[]>({
    queryKey: ["/attendancerecords/tree", "AttendanceApi"],
    queryFn: async () => ApiHelper.get("/attendancerecords/tree", "AttendanceApi")
    placeholderData: [],
  });
```

the queryKey is the cache key for this request. It must be an array but you can pass objects or subarrays to it. It stably converts it to a string that is compared to the previous queryKey. if it changes then the query will rerun.